core/types/json.go

```diff
@@ -44,6 +51,13 @@ func (b *hexBytes) UnmarshalJSON(input []byte) error {
 
 type hexBig big.Int
 
+func (b *hexBig) MarshalJSON() ([]byte, error) {
```

Comment:
@bas-vk 0X should be addressed, but it's part of an older PR I have that I'd really need to polish up. @fjl Because we currently consider it invalid, whereas it's valid hex. Also there are a few places in our code which don't accept capitalized hex numbers which could cause weird issues.